// react-canvas-confetti/dist/presets/realistic/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var index_1 = __importDefault(require("../index"));
var getTickAnimation = function (confetti, decorateOptions) {
    confetti(decorateOptions({
        spread: 26,
        startVelocity: 55,
        origin: { y: 0.7 },
        particleCount: Math.floor(200 * 0.25),
    }));
    confetti(decorateOptions({
        spread: 60,
        origin: { y: 0.7 },
        particleCount: Math.floor(200 * 0.2),
    }));
    confetti(decorateOptions({
        spread: 100,
        decay: 0.91,
        scalar: 0.8,
        origin: { y: 0.7 },
        particleCount: Math.floor(200 * 0.35),
    }));
    confetti(decorateOptions({
        spread: 120,
        startVelocity: 25,
        decay: 0.92,
        scalar: 1.2,
        origin: { y: 0.7 },
        particleCount: Math.floor(200 * 0.1),
    }));
    confetti(decorateOptions({
        spread: 120,
        startVelocity: 45,
        origin: { y: 0.7 },
        particleCount: Math.floor(200 * 0.1),
    }));
};
function Realistic(props) {
    return react_1.default.createElement(index_1.default, __assign({ getTickAnimation: getTickAnimation }, props));
}
exports.default = Realistic;
